docs/src/index.md

```diff
@@ -30,21 +30,17 @@ Some aspects of this process are documented [here](https://docs.julialang.org/en
 
 Every time you load a package in a fresh Julia session, the methods you use
 need to be JIT-compiled, and this contributes to the latency of using the package.
-In some circumstances, you can partially cache the results of compilation to a file
-(the `*.ji` files that live in your `~/.julia/compiled` directory) to reduce the burden
+In some circumstances, you can cache the results of compilation to files
+(the `*.ji` and `*.so` files that live in your `~/.julia/compiled` directory) to reduce the burden
 when your package is used.
```

Comment:
The parenthetical here breaks up the sentence flow and makes the idea harder to understand, so I moved this to the next sentence. `~/.julia/compiled` is the default location on most systems, but generally we need to refer to the Julia depot.

I then added some text to explain the trade off between caching more code and taking time to load it.

```suggestion
In some circumstances, you can cache the results of compilation to files to reduce the latency when your package is used. These files are the the `*.ji` and `*.so` files that live in the `compiled` directory of the Julia depot, usually located at `~/.julia/compiled`. However, if these files become large, loading them can be another source for latency. Julia needs time both to load and validate the cached compiled code. Minimizing the latency of using a package involves focusing on caching the compilation of code that is both commonly used and takes time to compile.
```

Reply:
Added in 503139b (manually, so I could insert line breaks :) ).